Nightly build failure, intel/19.0.5, intel/2021.3 (icpc) with new TestMDRangePolicyCTAD test

**Describe the bug**

The intel/19.0.5 nightly builds are failing with the recent #5516  ctad updates:

```
14:28:44 /home/jenkins/node/workspace/kokkos_osrhel7_intel_1905_openmp_pthread_serial/kokkos/core/unit_test/TestMDRangePolicyCTAD.cpp(86): error #177: member "<unnamed>::TestMDRangePolicyCTAD::i64" was declared but never referenced
14:28:44     [[maybe_unused]] static inline int64_t i64;
14:28:44                                            ^
14:28:44 
14:28:44 compilation aborted for /home/jenkins/node/workspace/kokkos_osrhel7_intel_1905_openmp_pthread_serial/kokkos/core/unit_test/TestMDRangePolicyCTAD.cpp (code 2)
14:28:44 make[2]: *** [core/unit_test/CMakeFiles/Kokkos_CoreTestCompileOnly.dir/TestMDRangePolicyCTAD.cpp.o] Error 2
14:28:44 make[2]: *** Waiting for unfinished jobs....
```

**Please include the following for a minimal reproducer**

**Reproducer**:
```bash
module load sems-cmake sems-intel/19.0.5

$KOKKOS_PATH/generate_makefile.bash --with-devices=Serial  --compiler=icpc --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="17" --kokkos-path=$KOKKOS_PATH --no-examples   --with-options=enable_large_mem_tests
```

--- a/core/unit_test/TestMDRangePolicyCTAD.cpp
+++ b/core/unit_test/TestMDRangePolicyCTAD.cpp
@@ -45,6 +45,7 @@ struct TestMDRangePolicyCTAD {
   [[maybe_unused]] static inline int64_t tt[5];
   [[maybe_unused]] static inline Kokkos::Array<int64_t, 3> a;
   [[maybe_unused]] static inline Kokkos::Array<int64_t, 2> aa;
+  [[maybe_unused]] static inline int64_t i64;
 
   // Workaround for nvc++ (CUDA-11.7-NVHPC) ignoring [[maybe_unused]] on
   // ImplicitlyConvertibleToDefaultExecutionSpace::operator
@@ -54,7 +55,7 @@ struct TestMDRangePolicyCTAD {
 
   // Workaround for HIP-ROCm-5.2 "declared but never referenced"
   TestMDRangePolicyCTAD() {
-    maybe_unused(des, notEs, ses, t, tt, a, aa, notEsToDes);
+    maybe_unused(des, notEs, ses, t, tt, a, aa, notEsToDes, i64);
   }
 
   // MDRangePolicy with C array parameters
@@ -83,7 +84,6 @@ struct TestMDRangePolicyCTAD {
                                decltype(Kokkos::MDRangePolicy(
                                    {1, 2, 3, 4, 5, 6}, {1, 2, 3, 4, 5, 6}))>);
 
-  [[maybe_unused]] static inline int64_t i64;
   static_assert(std::is_same_v<Kokkos::MDRangePolicy<Kokkos::Rank<6>>,
                                decltype(Kokkos::MDRangePolicy(
                                    {1, 2, 3, 4, 5, 6}, {1, 2, 3, 4, 5, 6},

